### ISSUE TITLE
[analyzer] Handle `[[assume(cond)]]` as `__builtin_assume(cond)`

### ISSUE THREAD

**Issue by steakhal** [llvm/llvm-project#100762]
We already handle `__builtin_assume` calls inside the `BuiltinFunctionChecker`.
We should also handle the attribute variant of this, e.g.:
https://compiler-explorer.com/z/oq6nY7eqE
```c++
void clang_analyzer_printState();
extern int arr[10];

void using_assume_attr(int x) {
    [[assume(x > 100)]]; // NullStmt with an attribute
    //clang_analyzer_printState();
    arr[x] = 404; // FIXME: we don't raise this!
}

void using_builtin(int x) {
    __builtin_assume(x > 100); // CallExpr
    //clang_analyzer_printState();
    arr[x] = 404; // caught!
}
```

**Comment by vortex73** [comment]
I'm working on this issue.

**Comment by duncpro** [comment]
I'll work on this issue.

**Comment by vinay-deshmukh** [comment]
Hi,
May I be assigned to this issue? I'd like to try to solve it!

**Comment by shrey1605** [comment]
Hi,
I would like to work on this. Can I please be assigned to this issue? Thank you.

**Comment by steakhal** [comment]
We usually don't assign issues. Contributors are expected to file a PR with their improvements when they have something.
